internal/hcl/graph.go

```diff
@@ -0,0 +1,325 @@
+package hcl
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/heimdalr/dag"
+	"github.com/sirupsen/logrus"
+	"github.com/zclconf/go-cty/cty"
+)
+
+var (
+	visitMu              sync.RWMutex
```

Comment:
This is a global mutext across all go routines. This means this will lock visits across projects that use completely different `Graph`s, meaning they will never hit concurrent read/write issue. We can change this to be a `Graph` scoped mutex, this should hopefully speed up the performance.